### ISSUE TITLE
Add support for additional format in choices option for StringField data model fields

### ISSUE THREAD

**Issue by esb** [foundryvtt/foundryvtt#11912]
### User Experience

Currently the ``choices`` for a ``StringField`` data model field can be a function, an Array of strings or an Object containing key => value elements.

I'd like to request that in addition the ``choices`` should be able to support an Array of [key, value] array elements.

The primary reason for this is that it makes sorting the choices easier as Array sorting is an intrinsic function. As the localization functions are not available at ``Init`` time, it is necessary to move any client language sorting of choices to run time with an ``choices`` function, and if the list is already an array, then the sorting becomes straightforward.

The current implementation forces the ``choices`` to be stored as an Object, which requires the Object to be converted to an Array, sorted and then converted back to an Object in order to provide client language based sorting for ``choices`` lists.

**Comment by aaclayton** [comment]
Will verify, but I think an array of objects is supported: `Array<{value: string, label: string}>`

**Comment by esb** [comment]
Of course, format I'm suggesting is also the natural output of ``Object.entries()``, so that seems a more natural fit.
